STYLE_GUIDE.md

```diff
@@ -0,0 +1,156 @@
+ * Every package should have a high level doc.go file to describe the purpose of that package, its main functions, and any other relevant information.
+ * `TODO` should not be used. If important enough should be recorded as an issue.
+ * `BUG` / `FIXME` should be used sparingly to guide future developers on some of the vulnerabilities of the code.
+ * `XXX` can be used in work-in-progress (prefixed with "WIP:" on github) branches but they must be removed before approving a PR.
+ * Applications (e.g. clis/servers) *should* panic on unexpected unrecoverable errors and print a stack trace.
+
+## Comments
+
+ * Use a space after comment deliminter (ex. `// your comment`).
+ * Many comments are not sentences, these should begin with a lower case letter and end without a period.
+ * The first letter of sentences in comments are capitalized and ends with a period.
+
+## Linters
+
+These must be applied to all (go) repos.
+
+ * [shellcheck](https://github.com/koalaman/shellcheck)
+ * [golangci (covers all important linters)](https://github.com/golangci/golangci-lint)
+   - See the `.golangcli.yml` file in each repo for linter configuration.
+
+## Various
+
+ * Reserve "Save" and "Load" for long-running persistence operations. When parsing bytes, use "Encode" or "Decode".
+ * Maintain consistency across the codebase.
+ * Functions that return functions should have the suffix `Fn`
+ * A struct generally shouldn’t have a field named after itself, aka. this shouldn't occur:
+``` golang
+type middleware struct {
+	middleware Middleware
+}
+```
+ * In comments, use "iff" to mean, "if and only if".
+ * Product names are capitalized, like "Tendermint", "Basecoin", "Protobuf", etc except in command lines: `tendermint --help`
+ * Acronyms are all capitalized, like "RPC", "gRPC", "API".  "MyID", rather than "MyId".
+ * Prefer errors.New() instead of fmt.Errorf() unless you're actually using the format feature with arguments. (otherwise it's needlessly inefficient, & it'll crap out on %*)
+
+## Importing Libraries
+
+Sometimes it's necessary to rename libraries to avoid naming collisions or ambiguity.
+
+ * Use [goimports](https://godoc.org/golang.org/x/tools/cmd/goimports)
+ * Separate imports into blocks - one for the standard lib, one for external libs and one for tendermint libs.
+ * Here are some common library labels for consistency:
+   - dbm "github.com/tendermint/tm-db"
+   - tmcmd "github.com/tendermint/tendermint/cmd/tendermint/commands"
+   - tmcfg "github.com/tendermint/tendermint/config/tendermint"
+   - tmtypes "github.com/tendermint/tendermint/types"
+ * Never use anonymous imports (the `.`), for example, `tmlibs/common` or anything else.
+ * When importing a pkg from the `tendermint/libs` directory, prefix the pkg alias with tm.
+     - tmbits "github.com/tendermint/tendermint/libs/bits"
+ * tip: Use the `_` library import to import a library for initialization effects (side effects)
+
+## Dependencies
+
```

Comment:
Maybe should say something about using modules here?

Reply:
There's an entire section about dependencies in the contributing file, perhaps we should just remove this completely

Reply:
Yeah, I agree, let's remove this